### PR TITLE
Catch the exception when pushing entity by non finite vector

### PR DESCRIPTION
**Pull request by 3meraldK**
#### Catching exception which would be thrown when interacting with zero length vectors (I guess?)
---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #5567

### CODE REVIEW

**Comment by Moderocky** [comment]
It would be better to check whether either (the value and direction) or (the mod length) are zero rather than trying it and catching a heavy exception.

If the result is zero for an unexpected reason we probably want to know about that so we can determine why rather than failing silently.

File: src/main/java/ch/njol/skript/effects/EffPush.java [hunk 1]
```diff
@@ -77,6 +77,10 @@ protected void execute(final Event e) {
 			final Vector mod = d.getDirection(en);
 			if (v != null)
 				mod.normalize().multiply(v.doubleValue());
+			if (!Double.isFinite(mod.getX()) || !Double.isFinite(mod.getY()) || !Double.isFinite(mod.getZ())) {
+				// Some component of the mod vector is not finite, so just stop
+				return;
+			}
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/effects/EffPush.java:83]:
```suggestion
			try {
				mod.checkFinite();
			} catch (IllegalArgumentException exception) {
				return;
			}
```
Spigot has a method for checking

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/effects/EffPush.java (hunk 1 above)
Comment (UnderscoreTud) [reply src/main/java/ch/njol/skript/effects/EffPush.java:83]: Throwing and catching an exception is slower than just checking, as java has to create a stacktrace every time and that takes resources. The current way is better

Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/effects/EffPush.java:83]: That's how Spigot implemented, but having a Spigot maintained method is better.

**Review by Moderocky — APPROVED** [review]

File: src/main/java/ch/njol/skript/effects/EffPush.java (hunk 1 above)
Comment (3meraldK) [reply src/main/java/ch/njol/skript/effects/EffPush.java:83]: As Moderocky said, handling an exception is a heavy task. Your suggestion is exactly what I did, and I got their feedback. I'm supporting the current way my pull handles the error.

File: src/main/java/ch/njol/skript/effects/EffPush.java [hunk 2]
```diff
@@ -77,6 +77,10 @@ protected void execute(final Event e) {
 			final Vector mod = d.getDirection(en);
 			if (v != null)
 				mod.normalize().multiply(v.doubleValue());
+			if (!Double.isFinite(mod.getX()) || !Double.isFinite(mod.getY()) || !Double.isFinite(mod.getZ())) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/effects/EffPush.java:80]:
```suggestion
			if (!(Double.isFinite(mod.getX()) && Double.isFinite(mod.getY()) && Double.isFinite(mod.getZ()))) {
```

**Review by AyhamAl-Ali — APPROVED** [review]

File: src/main/java/ch/njol/skript/effects/EffPush.java (hunk 2 above)
Comment (Moderocky) [reply src/main/java/ch/njol/skript/effects/EffPush.java:80]: Well done for spotting, this is the more efficient way to check it. :)

**Pull request merged**